src/crystal/system/win32/hostname.cr

```diff
@@ -0,0 +1,12 @@
+require "c/sysinfoapi"
+
+module Crystal::System
+  def self.hostname
+    name_size = LibC::MAX_COMPUTER_NAME_SIZE.to_u32
+    unless LibC.GetComputerNameExW(LibC::COMPUTER_NAME_FORMAT::ComputerNameDnsHostname, out machine_name, pointerof(name_size))
+      raise WinError.new("Failed to get machine hostname")
```

Comment:
`WinError` and `Errno` usually only receive the name of the failing function as message (= `GetComputerNameExW`).

Reply:
Will change this around.

Reply:
Fixed [here](https://github.com/crystal-lang/crystal/blob/b9b590442aa9d0afcce2cb16af0e65e8a846ae76/src/crystal/system/win32/hostname.cr).